### DISCUSSION TITLE
Limitation with Parameter 'k'

### DISCUSSION THREAD

**Discussion by vietpham1911** [stanford-futuredata/ColBERT#202] [Q&A]
I am using ColBERT for passage retrieval in my question-answering system and I have encountered a limitation that I would like to understand better:

I have set the parameter k to 1000, for getting the top 1000 passages. However, I have noticed that only the top 64 passages are returned instead of the expected 1000 even though ColBERT was going through over one million passages. Could you please help me understand the reason behind this limitation? You also can see it in the picture below.
![image](https://github.com/stanford-futuredata/ColBERT/assets/56722449/e8c3465d-5f07-4f4c-8f02-6a400fc83846)

**Comment by okhat** [comment]
This is indeed really weird. It should return more. But I’m guessing you may have to tweak the parameters ncells (and others) in the notebook. Increase ncells to 8 or even 16.

**Reply by vietpham1911** [reply]
Now, it is returning the top 1000 passages even though I did not change anything... Thanks for your reply!
